Add the RMI-KB TKL-F13HS

ramonimbao opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

Adds the TKL-F13HS PCB. Leverages VIA custom UI code for RGB indicator control.

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

drashna commented on keyboards/rmi_kb/tkl_f13hs/config.h:
```diff
@@ -0,0 +1,101 @@
+
+#pragma once
+
+/* VIA custom menu*/
+#define VIA_EEPROM_CUSTOM_CONFIG_SIZE 8
```
Should be fine, as is.  Sometimes people enabled via on their keymaps, and this would help keep it consistent.

drashna commented on keyboards/rmi_kb/tkl_f13hs/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,743 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "via.h"
+#include "eeprom_driver.h"
```
Neither of these should be needed.
```suggestion
```

drashna commented on keyboards/rmi_kb/tkl_f13hs/readme.md:
```diff
@@ -0,0 +1,18 @@
+# TKL-F13HS
+
+![TKL-F13HS](https://i.imgur.com/6kBCSh4l.png)
```
```suggestion
![TKL-F13HS](https://i.imgur.com/6kBCSh4h.png)
```

drashna commented on keyboards/rmi_kb/tkl_f13hs/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,743 @@
+    }
+}
+
+void via_custom_value_command_kb(uint8_t *data, uint8_t length) {
```
Honestly, all of this should probably be handled at the keyboard level, not the keymap level.

ramonimbao replied:
Not sure I follow. There's no other way to set which RGB LED gets triggered with what other than through the VIA custom menus, so I it belonged here.

ramonimbao commented on keyboards/rmi_kb/tkl_f13hs/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,69 @@
+};
+
+#ifdef RGB_MATRIX_ENABLE
+led_config_t g_led_config = {{
+    // Key Matrix to LED Index
+    {      2, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED,      1, NO_LED,      0, NO_LED },
+    { NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED },
+    { NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED },
+    {      3, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED },
+    { NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED },
+    { NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED, NO_LED }
+}, {
+    // LED Index to Physical Position
+    { 211, 0 }, { 182, 0 }, { 0, 0 }, { 2, 27 }
+}, {
+    // LED Index to Flag
+    LED_FLAG_INDICATOR, LED_FLAG_INDICATOR, LED_FLAG_INDICATOR, LED_FLAG_INDICATOR
+}};
```
Thanks! I just need to copy the relevant bits right? Or do I grab whatever the output of the command is and replace all of my `info.json` with it?

ramonimbao commented:
I've updated my code!

- Moved the LED config from the keymap to keyboard
- Moved the default RGB behavior from the default keymap to keyboard as per lesshonor's review
- Moved the VIA custom UI handling thing to keyboard as per drashna's review

ramonimbao commented on keyboards/rmi_kb/tkl_f13hs/config.h:
```diff
@@ -0,0 +1,54 @@
+/* RGB matrix */
+#define WS2812_PIO_USE_PIO1 // Force the usage of PIO1 peripheral, by default the WS2812 implementation uses the PIO0 peripheral
+#ifdef RGB_MATRIX_ENABLE
+#define RGB_MATRIX_LED_COUNT 4
```
Oh, alright. That's noted

tzarc commented:
Placing this PR on hold for the moment until we can get a bit of discussion going w.r.t. the prevalence of this VIA v3 situation and the relatively unmaintainable code.

ramonimbao commented:
Is there anything I need to do?

dunk2k commented on keyboards/rmi_kb/tkl_f13hs/info.json:
```diff
@@ -0,0 +1,135 @@
+            {"flags": 8, "matrix": [3, 0], "x": 2, "y": 27}
+        ]
+    },
+    "layouts": {
```
Looking at the image of PCB provided in `readme.md`, it appears this PCB supports a non-split right shift. Could you add the non-split right shift layout to `layouts`.

dunk2k commented on keyboards/rmi_kb/tkl_f13hs/rules.mk:
```diff
@@ -0,0 +1,3 @@
+WS2812_DRIVER = vendor
```
migrate to `info.json`
```suggestion
```

dunk2k commented on keyboards/rmi_kb/tkl_f13hs/info.json:
```diff
@@ -0,0 +1,135 @@
+            {"flags": 8, "matrix": [0, 0], "x": 0, "y": 0},
+            {"flags": 8, "matrix": [3, 0], "x": 2, "y": 27}
+        ]
+    },
```
migrated from `rules.mk`
```suggestion
    },
    "ws2812": {
        "driver": "vendor"
    },
```

dunk2k commented on keyboards/rmi_kb/tkl_f13hs/readme.md:
can method(s) for accessing bootloader be added please?

dunk2k commented on keyboards/rmi_kb/tkl_f13hs/readme.md:
```diff
@@ -0,0 +1,18 @@
+![TKL-F13HS](https://i.imgur.com/6kBCSh4h.png)
+
+* Keyboard Maintainer: [Ramon Imbao](https://github.com/ramonimbao)
+* Hardware Supported: RP2040
```
```suggestion
* Hardware Supported: TKL-F13HS PCB
```

dunk2k commented on keyboards/rmi_kb/tkl_f13hs/info.json:
```diff
@@ -0,0 +1,135 @@
+
+                {"matrix": [3, 0], "x":0, "y":3.25, "w":1.75},
+                {"matrix": [3, 1], "x":1.75, "y":3.25},
+                {"matrix": [3, 2], "x":2.75, "y":3.25},
+                {"matrix": [3, 3], "x":3.75, "y":3.25},
+                {"matrix": [3, 4], "x":4.75, "y":3.25},
+                {"matrix": [3, 5], "x":5.75, "y":3.25},
+                {"matrix": [3, 6], "x":6.75, "y":3.25},
+                {"matrix": [3, 7], "x":7.75, "y":3.25},
+                {"matrix": [3, 8], "x":8.75, "y":3.25},
+                {"matrix": [3, 9], "x":9.75, "y":3.25},
+                {"matrix": [3,10], "x":10.75, "y":3.25},
+                {"matrix": [3,11], "x":11.75, "y":3.25},
+                {"matrix": [3,12], "x":12.75, "y":3.25, "w":2.25},
+
+                {"matrix": [4, 0], "x":0, "y":4.25, "w":2.25},
+                {"matrix": [4, 2], "x":2.25, "y":4.25},
+                {"matrix": [4, 3], "x":3.25, "y":4.25},
+                {"matrix": [4, 4], "x":4.25, "y":4.25},
+                {"matrix": [4, 5], "x":5.25, "y":4.25},
+                {"matrix": [4, 6], "x":6.25, "y":4.25},
+                {"matrix": [4, 7], "x":7.25, "y":4.25},
+                {"matrix": [4, 8], "x":8.25, "y":4.25},
+                {"matrix": [4, 9], "x":9.25, "y":4.25},
+                {"matrix": [4,10], "x":10.25, "y":4.25},
+                {"matrix": [4,11], "x":11.25, "y":4.25},
+                {"matrix": [4,12], "x":12.25, "y":4.25, "w":1.75},
+                {"matrix": [4,13], "x":14, "y":4.25},
+                {"matrix": [4,15], "x":16.25, "y":4.25},
+
+                {"matrix": [5, 0], "x":0, "y":5.25, "w":1.5},
+                {"matrix": [5, 1], "x":1.5, "y":5.25},
+                {"matrix": [5, 2], "x":2.5, "y":5.25, "w":1.5},
+                {"matrix": [5, 7], "x":4, "y":5.25, "w":7},
+                {"matrix": [5,11], "x":11, "y":5.25, "w":1.5},
+                {"matrix": [5,12], "x":12.5, "y":5.25},
+                {"matrix": [5,13], "x":13.5, "y":5.25, "w":1.5},
+                {"matrix": [5,14], "x":15.25, "y":5.25},
+                {"matrix": [5,15], "x":16.25, "y":5.25},
+                {"matrix": [5,16], "x":17.25, "y":5.25}
```
```suggestion
    "community_layouts": ["tkl_f13_ansi_tsangan"],
    "layouts": {
        "LAYOUT_all": {
            "layout": [
                {"matrix": [0, 0], "x":0, "y":0},
                {"matrix": [0, 1], "x":1.25, "y":0},
                {"matrix": [0, 2], "x":2.25, "y":0},
                {"matrix": [0, 3], "x":3.25, "y":0},
                {"matrix": [0, 4], "x":4.25, "y":0},
                {"matrix": [0, 5], "x":5.5, "y":0},
                {"matrix": [0, 6], "x":6.5, "y":0},
                {"matrix": [0, 7], "x":7.5, "y":0},
                {"matrix": [0, 8], "x":8.5, "y":0},
                {"matrix": [0, 9], "x":9.75, "y":0},
                {"matrix": [0,10], "x":10.75, "y":0},
                {"matrix": [0,11], "x":11.75, "y":0},
                {"matrix": [0,12], "x":12.75, "y":0},
                {"matrix": [0,13], "x":14, "y":0},
                {"matrix": [0,14], "x":15.25, "y":0},
                {"matrix": [0,15], "x":16.25, "y":0},
                {"matrix": [0,16], "x":17.25, "y":0},

                {"matrix": [1, 0], "x":0, "y":1.25},
                {"matrix": [1, 1], "x":1, "y":1.25},
                {"matrix": [1, 2], "x":2, "y":1.25},
                {"matrix": [1, 3], "x":3, "y":1.25},
                {"matrix": [1, 4], "x":4, "y":1.25},
                {"matrix": [1, 5], "x":5, "y":1.25},
                {"matrix": [1, 6], "x":6, "y":1.25},
                {"matrix": [1, 7], "x":7, "y":1.25},
                {"matrix": [1, 8], "x":8, "y":1.25},
                {"matrix": [1, 9], "x":9, "y":1.25},
                {"matrix": [1,10], "x":10, "y":1.25},
                {"matrix": [1,11], "x":11, "y":1.25},
                {"matrix": [1,12], "x":12, "y":1.25},
                {"matrix": [1,13], "x":13, "y":1.25, "w":2},
                {"matrix": [1,14], "x":15.25, "y":1.25},
                {"matrix": [1,15], "x":16.25, "y":1.25},
                {"matrix": [1,16], "x":17.25, "y":1.25},

                {"matrix": [2, 0], "x":0, "y":2.25, "w":1.5},
                {"matrix": [2, 1], "x":1.5, "y":2.25},
                {"matrix": [2, 2], "x":2.5, "y":2.25},
                {"matrix": [2, 3], "x":3.5, "y":2.25},
                {"matrix": [2, 4], "x":4.5, "y":2.25},
                {"matrix": [2, 5], "x":5.5, "y":2.25},
                {"matrix": [2, 6], "x":6.5, "y":2.25},
                {"matrix": [2, 7], "x":7.5, "y":2.25},
                {"matrix": [2, 8], "x":8.5, "y":2.25},
                {"matrix": [2, 9], "x":9.5, "y":2.25},
                {"matrix": [2,10], "x":10.5, "y":2.25},
                {"matrix": [2,11], "x":11.5, "y":2.25},
                {"matrix": [2,12], "x":12.5, "y":2.25},
                {"matrix": [2,13], "x":13.5, "y":2.25, "w":1.5},
                {"matrix": [2,14], "x":15.25, "y":2.25},
                {"matrix": [2,15], "x":16.25, "y":2.25},
                {"matrix": [2,16], "x":17.25, "y":2.25},

                {"matrix": [3, 0], "x":0, "y":3.25, "w":1.75},
                {"matrix": [3, 1], "x":1.75, "y":3.25},
                {"matrix": [3, 2], "x":2.75, "y":3.25},
                {"matrix": [3, 3], "x":3.75, "y":3.25},
                {"matrix": [3, 4], "x":4.75, "y":3.25},
                {"matrix": [3, 5], "x":5.75, "y":3.25},
                {"matrix": [3, 6], "x":6.75, "y":3.25},
                {"matrix": [3, 7], "x":7.75, "y":3.25},
                {"matrix": [3, 8], "x":8.75, "y":3.25},
                {"matrix": [3, 9], "x":9.75, "y":3.25},
                {"matrix": [3,10], "x":10.75, "y":3.25},
                {"matrix": [3,11], "x":11.75, "y":3.25},
                {"matrix": [3,12], "x":12.75, "y":3.25, "w":2.25},

                {"matrix": [4, 0], "x":0, "y":4.25, "w":2.25},
                {"matrix": [4, 2], "x":2.25, "y":4.25},
                {"matrix": [4, 3], "x":3.25, "y":4.25},
                {"matrix": [4, 4], "x":4.25, "y":4.25},
                {"matrix": [4, 5], "x":5.25, "y":4.25},
                {"matrix": [4, 6], "x":6.25, "y":4.25},
                {"matrix": [4, 7], "x":7.25, "y":4.25},
                {"matrix": [4, 8], "x":8.25, "y":4.25},
                {"matrix": [4, 9], "x":9.25, "y":4.25},
                {"matrix": [4,10], "x":10.25, "y":4.25},
                {"matrix": [4,11], "x":11.25, "y":4.25},
                {"matrix": [4,12], "x":12.25, "y":4.25, "w":1.75},
                {"matrix": [4,13], "x":14, "y":4.25},
                {"matrix": [4,15], "x":16.25, "y":4.25},

                {"matrix": [5, 0], "x":0, "y":5.25, "w":1.5},
                {"matrix": [5, 1], "x":1.5, "y":5.25},
                {"matrix": [5, 2], "x":2.5, "y":5.25, "w":1.5},
                {"matrix": [5, 7], "x":4, "y":5.25, "w":7},
                {"matrix": [5,11], "x":11, "y":5.25, "w":1.5},
                {"matrix": [5,12], "x":12.5, "y":5.25},
                {"matrix": [5,13], "x":13.5, "y":5.25, "w":1.5},
                {"matrix": [5,14], "x":15.25, "y":5.25},
                {"matrix": [5,15], "x":16.25, "y":5.25},
                {"matrix": [5,16], "x":17.25, "y":5.25}
            ]
        },
        "LAYOUT_tkl_f13_ansi_tsangan": {
            "layout": [
                {"matrix": [0, 0], "x":0, "y":0},
                {"matrix": [0, 1], "x":1.25, "y":0},
                {"matrix": [0, 2], "x":2.25, "y":0},
                {"matrix": [0, 3], "x":3.25, "y":0},
                {"matrix": [0, 4], "x":4.25, "y":0},
                {"matrix": [0, 5], "x":5.5, "y":0},
                {"matrix": [0, 6], "x":6.5, "y":0},
                {"matrix": [0, 7], "x":7.5, "y":0},
                {"matrix": [0, 8], "x":8.5, "y":0},
                {"matrix": [0, 9], "x":9.75, "y":0},
                {"matrix": [0,10], "x":10.75, "y":0},
                {"matrix": [0,11], "x":11.75, "y":0},
                {"matrix": [0,12], "x":12.75, "y":0},
                {"matrix": [0,13], "x":14, "y":0},
                {"matrix": [0,14], "x":15.25, "y":0},
                {"matrix": [0,15], "x":16.25, "y":0},
                {"matrix": [0,16], "x":17.25, "y":0},

                {"matrix": [1, 0], "x":0, "y":1.25},
                {"matrix": [1, 1], "x":1, "y":1.25},
                {"matrix": [1, 2], "x":2, "y":1.25},
                {"matrix": [1, 3], "x":3, "y":1.25},
                {"matrix": [1, 4], "x":4, "y":1.25},
                {"matrix": [1, 5], "x":5, "y":1.25},
                {"matrix": [1, 6], "x":6, "y":1.25},
                {"matrix": [1, 7], "x":7, "y":1.25},
                {"matrix": [1, 8], "x":8, "y":1.25},
                {"matrix": [1, 9], "x":9, "y":1.25},
                {"matrix": [1,10], "x":10, "y":1.25},
                {"matrix": [1,11], "x":11, "y":1.25},
                {"matrix": [1,12], "x":12, "y":1.25},
                {"matrix": [1,13], "x":13, "y":1.25, "w":2},
                {"matrix": [1,14], "x":15.25, "y":1.25},
                {"matrix": [1,15], "x":16.25, "y":1.25},
                {"matrix": [1,16], "x":17.25, "y":1.25},

                {"matrix": [2, 0], "x":0, "y":2.25, "w":1.5},
                {"matrix": [2, 1], "x":1.5, "y":2.25},
                {"matrix": [2, 2], "x":2.5, "y":2.25},
                {"matrix": [2, 3], "x":3.5, "y":2.25},
                {"matrix": [2, 4], "x":4.5, "y":2.25},
                {"matrix": [2, 5], "x":5.5, "y":2.25},
                {"matrix": [2, 6], "x":6.5, "y":2.25},
                {"matrix": [2, 7], "x":7.5, "y":2.25},
                {"matrix": [2, 8], "x":8.5, "y":2.25},
                {"matrix": [2, 9], "x":9.5, "y":2.25},
                {"matrix": [2,10], "x":10.5, "y":2.25},
                {"matrix": [2,11], "x":11.5, "y":2.25},
                {"matrix": [2,12], "x":12.5, "y":2.25},
                {"matrix": [2,13], "x":13.5, "y":2.25, "w":1.5},
                {"matrix": [2,14], "x":15.25, "y":2.25},
                {"matrix": [2,15], "x":16.25, "y":2.25},
                {"matrix": [2,16], "x":17.25, "y":2.25},

                {"matrix": [3, 0], "x":0, "y":3.25, "w":1.75},
                {"matrix": [3, 1], "x":1.75, "y":3.25},
                {"matrix": [3, 2], "x":2.75, "y":3.25},
                {"matrix": [3, 3], "x":3.75, "y":3.25},
                {"matrix": [3, 4], "x":4.75, "y":3.25},
                {"matrix": [3, 5], "x":5.75, "y":3.25},
                {"matrix": [3, 6], "x":6.75, "y":3.25},
                {"matrix": [3, 7], "x":7.75, "y":3.25},
                {"matrix": [3, 8], "x":8.75, "y":3.25},
                {"matrix": [3, 9], "x":9.75, "y":3.25},
                {"matrix": [3,10], "x":10.75, "y":3.25},
                {"matrix": [3,11], "x":11.75, "y":3.25},
                {"matrix": [3,12], "x":12.75, "y":3.25, "w":2.25},

                {"matrix": [4, 0], "x":0, "y":4.25, "w":2.25},
                {"matrix": [4, 2], "x":2.25, "y":4.25},
                {"matrix": [4, 3], "x":3.25, "y":4.25},
                {"matrix": [4, 4], "x":4.25, "y":4.25},
                {"matrix": [4, 5], "x":5.25, "y":4.25},
                {"matrix": [4, 6], "x":6.25, "y":4.25},
                {"matrix": [4, 7], "x":7.25, "y":4.25},
                {"matrix": [4, 8], "x":8.25, "y":4.25},
                {"matrix": [4, 9], "x":9.25, "y":4.25},
                {"matrix": [4,10], "x":10.25, "y":4.25},
                {"matrix": [4,11], "x":11.25, "y":4.25},
                {"matrix": [4,12], "x":12.25, "y":4.25, "w":2.75},
                {"matrix": [4,15], "x":16.25, "y":4.25},

                {"matrix": [5, 0], "x":0, "y":5.25, "w":1.5},
                {"matrix": [5, 1], "x":1.5, "y":5.25},
                {"matrix": [5, 2], "x":2.5, "y":5.25, "w":1.5},
                {"matrix": [5, 7], "x":4, "y":5.25, "w":7},
                {"matrix": [5,11], "x":11, "y":5.25, "w":1.5},
                {"matrix": [5,12], "x":12.5, "y":5.25},
                {"matrix": [5,13], "x":13.5, "y":5.25, "w":1.5},
                {"matrix": [5,14], "x":15.25, "y":5.25},
                {"matrix": [5,15], "x":16.25, "y":5.25},
                {"matrix": [5,16], "x":17.25, "y":5.25}
```

dunk2k commented on keyboards/rmi_kb/tkl_f13hs/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,27 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
```
```suggestion
    [0] = LAYOUT_all(
```

dunk2k commented on keyboards/rmi_kb/tkl_f13hs/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,51 @@
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_F13,  KC_PSCR, KC_SCRL, KC_PAUS,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),                     KC_UP,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      KC_RALT, KC_RGUI, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,
+        _______, _______, _______,                            _______,                                     _______, _______, _______, _______, _______, _______
+    ),
+    [2] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,
+        _______, _______, _______,                            _______,                                     _______, _______, _______, _______, _______, _______
+    ),
+    [3] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,
+        _______, _______, _______,                            _______,                                     _______, _______, _______, _______, _______, _______
+    )
```
```suggestion
    [0] = LAYOUT_all(
        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_F13,  KC_PSCR, KC_SCRL, KC_PAUS,
        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_INS,  KC_HOME, KC_PGUP,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END,  KC_PGDN,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),                     KC_UP,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                                      KC_RALT, KC_RGUI, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
    ),
    [1] = LAYOUT_all(
        QK_BOOT, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,
        _______, _______, _______,                            _______,                                     _______, _______, _______, _______, _______, _______
    ),
```
